### PR TITLE
rewired the api calls to happen on the dashboard

### PR DESCRIPTION
**Pull request by L-Steinmacher**
**Description**
I removed the api calls on the components and put it on the dashboard to allow other components to use the state.
**What work was done?**  
Fixed context and actions and rewired the api call.  
**Why was this work done?**  
Rewiring  on the page render allows many components to use the state. 

Please delete options that are not relevant.  
[] New feature (non-breaking change which adds functionality)
[x] Bug fix
Change Status  
[x] Completed, ready to review and merge  
Has This Been Tested  
[x] Yes
Checklist
[x] My code follows the style guidelines of this project  
[x] I have performed a self-review of my own code  
[] My code has been reviewed by at least one peer  
[] I have commented my code, particularly in hard-to-understand areas  
[] I have made corresponding changes to the documentation  
[x] My changes generate no new warnings  
[x] There are no merge conflicts

### CODE REVIEW

File: src/components/pages/Clubs/ViewClubs.js [hunk 1]
```diff
@@ -35,22 +35,21 @@ function ViewClubs(props) {
   const [tableData, setTableData] = useState(sampleTableData);
   const context = useContext(AdminContext);
 
-  // Get activities and set to context
+  /*
+   * Makes axios call if the context is empty when the page is reloaded.
+   */
   useEffect(() => {
-    fetchClubs();
+    if (context.clubs.length === 0) {
```
Comment (eet349) [inline src/components/pages/Clubs/ViewClubs.js:42]: Good job accounting for edge cases

File: src/components/pages/Clubs/ViewClubs.js [hunk 2]
```diff
@@ -73,7 +72,7 @@ function ViewClubs(props) {
             title={'Clubs'}
             subTitle={`Sorted by clubs`}
           />
-          <ImportClubs fetchClubs={fetchClubs} />
+          <ImportClubs fetchClubs={() => getClubs('authState', context)} />
```
Comment (eet349) [inline src/components/pages/Clubs/ViewClubs.js:75]: Good job keeping the work on this feature within scope and not digging into legacy code

File: src/components/pages/DashboardAlerts/RenderDashboardAlerts.js [hunk 3]
```diff
@@ -20,46 +21,15 @@ function RenderDashboardAlerts() {
     const rtn = {};
 
     arr.forEach(alert => {
-      if (!(alert.clubactivity.club.clubname in rtn)) {
-        rtn[alert.clubactivity.club.clubname] = 0;
+      if (!(alert.clubname in rtn)) {
```
Comment (eet349) [inline src/components/pages/DashboardAlerts/RenderDashboardAlerts.js:24]: Thank you for changing this to reflect the changes in the backend.

**Review by eet349 — APPROVED** [review]
Good job, you've made a lot of progress on this feature

**Review by ashtilawat23 — APPROVED** [review]

**Pull request merged**